ArcGISRuntimeSDKQt_CppSamples/UtilityNetwork/ValidateUtilityNetworkTopology/ValidateUtilityNetworkTopology.cpp

```diff
@@ -0,0 +1,671 @@
+  const Envelope m_extent = m_mapView->currentViewpoint(ViewpointType::BoundingGeometry).targetGeometry().extent();
+
+  m_busy = true;
+  emit isBusy();
+
+  m_message = QString("Validating utility network topology...");
+  emit messageChanged();
+
+  // Get the validation result
+  UtilityNetworkValidationJob* job = m_utilityNetwork->validateNetworkTopology(m_extent);
+  job->start();
+
+  QObject::connect(job, &UtilityNetworkValidationJob::statusChanged, [this, job](JobStatus status)
+  {
+    if (status == JobStatus::Succeeded)
+    {
+      UtilityValidationResult* result = job->result();
+      result->hasDirtyAreas();
+      result->hasErrors();
+
+      m_message = QString("Utility Validation Result:\n"
+          "    Has Dirty Areas: " + QString(result->hasDirtyAreas() ? "true" : "false") + "\n"
+          "    Has Errors: " + QString(result->hasErrors() ? "true" : "false") + "\n"
+          "    Click 'Get State' to check the updated network state.");
+      emit messageChanged();
+
+      m_validateBtn = result->hasDirtyAreas();
+      emit validateBtn();
+
+      m_busy = false;
+      emit isBusy();
+    }
+    else
+    {
+      m_message = QString("Validate network topology failed.");
+      emit messageChanged();
+
+      m_busy = false;
+      emit isBusy();
+    }
+  });
+}
+
+void ValidateUtilityNetworkTopology::onTrace()
+{
+  m_utilityNetwork = m_mapView->map()->utilityNetworks()->first();
+
+  m_message = QString("Running a downstream trace...");
+  emit messageChanged();
+
+  m_busy = true;
+  emit isBusy();
+
+  // Clear previous selection from the layers.
+  m_deviceFeatureLayer->clearSelection();
+  m_lineFeatureLayer->clearSelection();
+
+  //  Get the trace result from the utility network.
+  m_utilityNetwork->traceAsync(m_traceParameters).then(this, [this](QList<UtilityTraceResult*>)
+  {
+
+    m_busy = false;
+    emit isBusy();
+
+    UtilityTraceResult* result = m_utilityNetwork->traceResult()->at(0);
+
+    const QList<UtilityElement*> elements = static_cast<UtilityElementTraceResult*>(result)->elements(this);
+    m_message = QString("Trace completed: %1 elements found.").arg(elements.length());
+    emit messageChanged();
+
+    QueryParameters deviceParams;
+    QueryParameters lineParams;
+    QList<qint64> deviceObjIds;
+    QList<qint64> lineObjIds;
+
+    for (UtilityElement* item : elements)
+    {
+      if (item->networkSource()->name() == "Electric Distribution Device")
+        deviceObjIds.append(item->objectId());
+      else if (item->networkSource()->name() == "Electric Distribution Line")
+        lineObjIds.append(item->objectId());
+    }
+
+    deviceParams.setObjectIds(deviceObjIds);
+    lineParams.setObjectIds(lineObjIds);
+
+    m_deviceFeatureLayer->selectFeaturesAsync(deviceParams, SelectionMode::Add).then(this, [](FeatureQueryResult*)
+    {
+    });
+
+    m_lineFeatureLayer->selectFeaturesAsync(lineParams, SelectionMode::Add).then(this, [](FeatureQueryResult*)
+    {
+    });
```

Comment:
This makes sense if you don't need to do anything here, but you could clean up unused memory here.

```suggestion
    m_deviceFeatureLayer->selectFeaturesAsync(deviceParams, SelectionMode::Add).then(this, [](FeatureQueryResult* rawResult)
    {
      std::unique_ptr<FeatureQueryResult> {rawResult};
    });

    m_lineFeatureLayer->selectFeaturesAsync(lineParams, SelectionMode::Add).then(this, [](FeatureQueryResult* rawResult)
    {
      std::unique_ptr<FeatureQueryResult> {rawResult};
    });
```

Reply:
updated